packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol

```diff
@@ -298,6 +318,12 @@ contract FuelERC20GatewayV4 is
         emit Withdrawal(bytes32(uint256(uint160(to))), tokenAddress, amount);
     }
 
+    /// @notice Checks if the rate limit is enabled or not
+    /// @param tokenAddress Address of the token to check rate limit status for
+    function isRateLimitEnabled(address tokenAddress) public view returns (bool) {
```

Comment:
nit: `rateLimitStatus` mapping is public. do you think we need this?

Reply:
just added for better readability that's the only reason

Reply:
I feel its redundant, but no harm in having it as well. feel free to keep it if you prefer.

Reply:
removed